sqlmesh/core/engine_adapter/fabric.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import typing as t
+from sqlglot import exp
+from sqlmesh.core.engine_adapter.mssql import MSSQLEngineAdapter
+from sqlmesh.core.engine_adapter.shared import InsertOverwriteStrategy, SourceQuery
+from sqlmesh.core.engine_adapter.base import EngineAdapter
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import TableName
+
+
+class FabricAdapter(MSSQLEngineAdapter):
```

Comment:
I added this to my own fork, and it solved the merge issue with incremental_by_unique_key:
```suggestion
from sqlmesh.core.engine_adapter.mixins import LogicalMergeMixin

class FabricAdapter(LogicalMergeMixin, MSSQLEngineAdapter):
```